lib/Documentable/Search.pm6

```diff
@@ -11,32 +12,42 @@ class Documentable::Search {
     method generate-entries($registry) {
         say "Generating entries";
         my @entries;
-        for Kind::Type, Kind::Language, Kind::Programs -> $kind {
-            @entries.append: $registry.lookup($kind.Str, :by<kind>).map(-> $doc {
-                self.search-entry(
-                    category => $doc.subkinds[0],
-                    value    => $doc.name,
-                    url      => $doc.url
-                )
-            }).Slip;
+        self.consume-entries-by-kind(@entries, $registry, Kind::Type, 'Types');
+        self.consume-entries-by-kind(@entries, $registry, Kind::Language, 'Language');
+        self.consume-entries-by-kind(@entries, $registry, Kind::Programs, 'Programs');
+
+        for Kind::Routine.Str -> $kind {
+            @entries.append: $registry.lookup($kind, :by<kind>)
+                    .categorize({ .name })
+                    .pairs.sort({ .key })
+                    .map(-> (:key($name), :value(@docs)) {
+                        my $category = self.calculate-category(@docs, $kind);
+                        self.search-entry(:$category,
+                                value => escape($name), url => escape-json("/{ $kind.lc }/{ good-name($name) }"));
+                    });
         }
 
-        for Kind::Routine, Kind::Syntax.Str -> $kind {
-            @entries.append:  $registry.lookup($kind, :by<kind>)
-                            .categorize({.name})
-                            .pairs.sort({.key})
-                            .map( -> (:key($name), :value(@docs)) {
-                                    self.search-entry(
-                                        category => @docs > 1 ?? $kind.gist !! @docs[0].subkinds[0] || $kind.gist,
-                                        value    => escape($name),
-                                        url      => escape-json("/{$kind.lc}/{good-name($name)}")
+        for Kind::Syntax.Str -> $kind {
```

Comment:
Would need to check line by line here, but it looks mostly repeated. And what's not, could probably be factored out to a different function.
Can't see how this is related to categories, except that now category is calculated instead of having a weird formula. That's nice.

Reply:
You are right, removed the copypaste.